pyresample/test/test_geometry/test_swath.py

```diff
@@ -34,8 +34,8 @@ def _gen_swath_def_xarray_dask(create_test_swath):
     lons, lats = _gen_swath_lons_lats()
     lons_dask = da.from_array(lons)
     lats_dask = da.from_array(lats)
-    lons_xr = xr.DataArray(lons_dask, dims=('y', 'x'))
-    lats_xr = xr.DataArray(lats_dask, dims=('y', 'x'))
+    lons_xr = xr.DataArray(lons_dask, dims=('y', 'x'), attrs={'name': 'Longitude', 'resolution': 500, 'units': 'm'})
+    lats_xr = xr.DataArray(lats_dask, dims=('y', 'x'), attrs={'name': 'Latitude', 'resolution': 500, 'units': 'm'})
```

Comment:
I'm a little scared about this. The resolution and units I feel like are not guaranteed even in Satpy...eh I guess they should be. I don't know if I'm scared enough (that this breaks checks that are needed) to ask you to change it. I guess leave it for now.

Reply:
Yes I agree. As far as I see this does not break things right now. But I could change it back and write extra Swath generation functions for the repr tests. Actually while adding this made me think again about the situation I described above.

So what I meant was that currently in the xarray case the representation relies on the name, resolution and units attributes being set in the `DataArray`. If that is not the case, for example a user just provides  plain `DataArrays` with lons/lats, the representation currently throws an error.
So either these can be made mandatory, which is a hard constraint for the `SwathDefinition` I guess or the representation gets some more checks to set/calculate these. I hope this makes it clearer.
In the numpy case this is already done now including a very rough calculation of the resolution (https://github.com/BENR0/pyresample/blob/67f9df2862a2527a8142a28f65ff436f65127d38/pyresample/_formatting_html.py#L243-L249).

Reply:
FYI the `resolution` in the DataArray is the nominal resolution in the `DataID` and doesn't necessarily match/reflect the current state of the DataArray. After resampling (which is not common for a target swath definition) the resolution could be completely different.